src/db.js

```diff
@@ -1,4 +1,87 @@
 // @flow
 import store from "react-native-simple-store";
 
-export default store;
+const ACCOUNTS_KEY = "accounts";
+const ACCOUNTS_DB_PREFIX = "accounts.active.";
+
+export default class DB {
+  static get(key: string | any[]): Promise<any> {
+    /** handle accounts case */
+    if (key === ACCOUNTS_KEY) return DB.getAccounts();
+    return store.get(key);
+  }
+
+  static save(key: string | any[], value: any): Promise<any> {
+    /** handle accounts case */
+    if (key === ACCOUNTS_KEY) return DB.saveAccounts(value);
+    return store.save(key, value);
+  }
+
+  static update(key: string, value: any): Promise<any> {
+    return store.update(key, value);
+  }
+
+  static keys(): Promise<any> {
+    return store.keys();
+  }
+
+  static delete(key: string | any[]): Promise<any> {
+    return store.delete(key);
+  }
+
+  static push(key: string, value: any): Promise<any> {
+    return store.push(key, value);
+  }
+
+  static getAccountDbKey = (id: string): string => `${ACCOUNTS_DB_PREFIX}${id}`;
+
+  /** save accounts method between SQLite db and redux store persist */
+  static saveAccounts({
+    active: newAccounts,
+  }: {
+    active: any[],
+  }): Promise<any> {
+    return DB.getAccounts().then(oldAccounts => {
+      let deletedKeys = [ACCOUNTS_KEY];
+      if (oldAccounts && oldAccounts.active) {
+        /** add old accounts removed from redux store to keys to remove from SQLite db */
+        deletedKeys = deletedKeys.concat(
+          oldAccounts.active
+            .filter(({ data }) =>
+              newAccounts.every(na => na.data.id !== data.id),
+            )
+            .map(({ data }) => DB.getAccountDbKey(data.id)),
+        );
+      }
+
+      /** format data for DB persist */
+      const dbData = newAccounts.map(({ data }) => [
+        DB.getAccountDbKey(data.id),
+        { data, version: 1 },
+      ]);
+
+      /** persist store data to DB */
+      return store.save(dbData).then(() => {
+        /** then delete potential removed keys */
+        DB.delete(deletedKeys);
+      });
+    });
+  }
+
+  /** get accounts specific method to agregate all account keys into the correct format */
+  static getAccounts(): Promise<any> {
+    /** fetch all DB keys */
+    return DB.keys().then(keys => {
+      /** filter through them to get only the accounts ones */
+      const accountKeys = keys.filter(
+        key => key.indexOf(ACCOUNTS_DB_PREFIX) === 0,
+      );
+      /** if present return them */
+      if (accountKeys && accountKeys.length > 0)
+        return store.get(accountKeys).then(active => ({ active }));
+
+      /** else return old accounts data */
+      return store.get(ACCOUNTS_KEY);
```

Comment:
is this resilient to potentially account key that is getting dropped? If not, we need to protect atomicity of that logic VS set accounts.